src/test/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceSourceConfigTest.java

```diff
@@ -274,4 +274,51 @@ private void testPKChunkInvalidConfig(SalesforceSourceConfig config, String stag
     }
     Assert.assertEquals(stageConfigName, failure.getCauses().get(0).getAttribute(CauseAttributes.STAGE_CONFIG));
   }
+
+  @Test
+  public void testValidateFilters() {
+    SalesforceSourceConfig config = new SalesforceSourceConfigBuilder()
+            .setQuery("")
+            .setSObjectName("Account")
+            .setDatetimeBefore("test")
+            .setDatetimeAfter("test")
+            .setDuration("5 days")
+            .setOffset("5 days")
+            .build();
+
+    MockFailureCollector collector = new MockFailureCollector();
+    SalesforceSourceConfig mock = Mockito.spy(config);
+    Mockito.when(mock.canAttemptToEstablishConnection()).thenReturn(false);
+    try {
+      mock.validate(collector);
+      Assert.assertEquals(2, collector.getValidationFailures().size());
+    } catch (ValidationException e) {
+      Assert.assertEquals(1, e.getFailures().size());
+    }
```

Comment:
fail if exception is not thrown

Reply:
On test re-run, it was observed that no exception could be thrown. So removed the `try-catch` block.